mongodbatlas/fw_resource_mongodbatlas_project_migration_test.go

```diff
@@ -0,0 +1,229 @@
+package mongodbatlas
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+)
+
+var _ plancheck.PlanCheck = debugPlan{}
+
+type debugPlan struct{}
+
+func (e debugPlan) CheckPlan(ctx context.Context, req plancheck.CheckPlanRequest, resp *plancheck.CheckPlanResponse) {
+	rd, err := json.Marshal(req.Plan)
+	if err != nil {
+		tflog.Debug(ctx, fmt.Sprintf("error marshaling machine-readable plan output: %s", err))
+	}
+	tflog.Info(ctx, fmt.Sprintf("req.Plan - %s\n", string(rd)))
+}
+
+func DebugPlan() plancheck.PlanCheck {
+	return debugPlan{}
+}
+
+func TestAccRSProject_Migration_NoProps(t *testing.T) {
+	var (
+		resourceName = "mongodbatlas_project.test"
+		projectName  = acctest.RandomWithPrefix("test-acc-migration")
+		orgID        = os.Getenv("MONGODB_ATLAS_ORG_ID")
+	)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckBasic(t) },
+		CheckDestroy: testAccCheckMongoDBAtlasProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"mongodbatlas": {
+						VersionConstraint: "1.11.0",
+						Source:            "mongodb/mongodbatlas",
+					},
+				},
+				Config: fmt.Sprintf(`resource "mongodbatlas_project" "test" {
+					name   = "%s"
+					org_id = "%s"
+				  }`, projectName, orgID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "org_id", orgID),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: testAccProviderV6Factories,
```

Comment:
how this step will work? is the first step using the old provider and the second step the framework?

Reply:
Correct. First step will use version 1.11.0 of our provider that was published today.
Second step will create a plan using the Framework-based implementation of this resource/provider and check that no changes should be discovered as part of the plan.